File: src/data/types/crewCapacity.ts
// Phase 4 GDD §2.4 -- a player's crew capacity: a small base plus
// purchasable expansion slots. Current crew count vs.
// (baseCapacity + purchasedSlots) is what hireCrew() checks against.
export interface CrewCapacity {
  playerId: string;
  baseCapacity: number;
  purchasedSlots: number;
}

File: unity/ProfitableCore/Schema/CrewCapacity.cs
namespace Profitable.Core.Schema;

// Ports src/data/types/crewCapacity.ts. A player's crew capacity: a small
// base plus purchasable expansion slots. Current crew count vs.
// (BaseCapacity + PurchasedSlots) is what HireCrew checks against.
public class CrewCapacity
{
    public string PlayerId { get; set; } = string.Empty;
    public int BaseCapacity { get; set; }
    public int PurchasedSlots { get; set; }
}
